File: aexample/src/main/java/org/judal/examples/java/jdbc/E07_ScanTablesFromExistingDatamodel.java
package org.judal.examples.java.jdbc;

import java.util.Map.Entry;

import org.junit.Test;
import org.judal.metadata.SchemaMetaData;
import org.judal.metadata.TableDef;
import org.judal.jdbc.JDBCRelationalDataSource;

/**
 * Example of how to read the metadata of tables
 * already present in a datamodel.
 */
public class E07_ScanTablesFromExistingDatamodel {

	@Test
	public void demo() throws Exception {

		// Simulate a data model that already exists
		JDBCRelationalDataSource dataSource = setUp();
		
		// Walk tables
		for (Entry<String,TableDef> entry : dataSource.getJDCTablesMap().entrySet())
			System.out.println("Scanned table " + entry.getValue().getName());

		// Walk schema objects
		SchemaMetaData schemaMetadata = dataSource.getMetaData();
		for (TableDef tableDef : schemaMetadata.tables())
			System.out.println("Scanned table " + tableDef.getName());
		
		tearDown (dataSource);
	}

	public static JDBCRelationalDataSource setUp() throws Exception {
		JDBCRelationalDataSource dataSource = E02_CreateAdditionalDataSource.create();
		E06_CreateTablesFromDDLXML.createSchemaObjects(dataSource);
		return dataSource;
	}

	public static void tearDown(JDBCRelationalDataSource dataSource) throws Exception {
		E06_CreateTablesFromDDLXML.dropSchemaObjects(dataSource);
		E02_CreateAdditionalDataSource.close(dataSource);
	}
	
}

File: aexample/src/main/scala/org/judal/examples/scala/jdbc/E07_ScanTablesFromExistingDatamodel.scala
package org.judal.examples.scala.jdbc

import org.junit.Test

import org.scalatest.Suite

import org.judal.jdbc.JDBCRelationalDataSource

import scala.collection.JavaConversions._

/**
 * Example of how to read the metadata of tables
 * already present in a datamodel.
 */
class E07_ScanTablesFromExistingDatamodel extends Suite {

	@Test def demo() {

		// Simulate a data model that already exists
		val dataSource = E07_ScanTablesFromExistingDatamodel.setUp()
		
		// Walk tables
		dataSource.getJDCTablesMap.entrySet.foreach { entry => println("Scanned table " + entry.getValue.getName) }

		// Walk schema objects
		dataSource.getMetaData.tables.foreach { tdef => println("Scanned table " + tdef.getName) }
		
		E07_ScanTablesFromExistingDatamodel.tearDown (dataSource)
	}
	
}

object E07_ScanTablesFromExistingDatamodel {

	def setUp() = {
		val dataSource = E02_CreateAdditionalDataSource.create()
		E06_CreateTablesFromDDLXML.createSchemaObjects(dataSource)
		dataSource
	}

	def tearDown(dataSource: JDBCRelationalDataSource) = {
		E06_CreateTablesFromDDLXML.dropSchemaObjects(dataSource)
		E02_CreateAdditionalDataSource.close(dataSource)
	}
  
}